src/main/java/uk/ac/ebi/eva/pipeline/io/writers/AnnotationInVariantMongoWriter.java

```diff
@@ -0,0 +1,173 @@
+        bulkPrepare(bulkOperations, variantAnnotations);
+        bulkOperations.execute();
+    }
+
+    private void bulkPrepare(BulkOperations bulkOperations, Map<String, VariantAnnotation> variantAnnotations) {
+        Map<String, VariantAnnotation> storedVariantAnnotations = getStoredVariantAnnotations(variantAnnotations);
+
+        for (Map.Entry<String, VariantAnnotation> entry : variantAnnotations.entrySet()) {
+            final String key = entry.getKey();
+            if (storedVariantAnnotations.containsKey(key)) {
+                bulkUpdate(bulkOperations, key, storedVariantAnnotations.get(key).concatenate(entry.getValue()));
+            } else {
+                bulkAddToSet(bulkOperations, key, entry.getValue());
+            }
+        }
+    }
+
+    private void bulkAddToSet(BulkOperations bulkOperations, String variantId, VariantAnnotation value) {
+        DBObject id = new BasicDBObject(ID, variantId);
+        DBObject variantAnnotation = convertToMongo(value);
+        BasicDBObject addToSet = new BasicDBObject(ADD_TO_SET, new BasicDBObject(ANNOTATION_FIELD, variantAnnotation));
+        bulkOperations.updateOne(new BasicQuery(id), new BasicUpdate(addToSet));
+    }
+
+    private void bulkUpdate(BulkOperations bulkOperations, String variantId, VariantAnnotation value) {
+        BasicDBObject query = generateQueryForAnnotationInVariant(variantId);
+
+        DBObject variantAnnotation = convertToMongo(value);
+        final BasicDBObject annotation = new BasicDBObject(ANNOTATION_IN_LIST, variantAnnotation);
+        BasicDBObject setAnnotation = new BasicDBObject(SET, annotation);
+
+        bulkOperations.updateOne(new BasicQuery(query), new BasicUpdate(setAnnotation));
+    }
+
+    private BasicDBObject generateQueryForAnnotationInVariant(String... variantIds) {
+        BasicDBObject query = new BasicDBObject();
+        if (variantIds.length == 1) {
+            query.append(ID, variantIds[0]);
+        } else {
+            query.append(ID, new BasicDBObject(IN, variantIds));
+        }
+        query.append(ANNOTATION_FIELD, createQueryMatchForVepAndCacheVersion());
+        return query;
+    }
+
+    private BasicDBObject createQueryMatchForVepAndCacheVersion() {
+        BasicDBObject annotationQuery = new BasicDBObject();
+        annotationQuery.append(VEP_VERSION_FIELD, vepVersion);
+        annotationQuery.append(VEP_CACHE_VERSION_FIELD, vepCacheVersion);
+        return new BasicDBObject(ELEM_MATCH, annotationQuery);
+    }
+
+    private Map<String, VariantAnnotation> getStoredVariantAnnotations(Map<String, VariantAnnotation> variantAnnotations) {
+        Map<String, VariantAnnotation> storedVariantAnnotations = new HashMap<>();
+        BasicDBObject query = generateQueryForAnnotationInVariant(variantAnnotations.keySet().toArray(new String[]{}));
+
+        Iterator<DBObject> iterator = mongoOperations.getCollection(collection).find(query).iterator();
```

Comment:
This is spring MongoOperations. That overloaded function doesn't exist like that.